h5io/_h5io.py

```diff
@@ -63,9 +63,21 @@ def _create_titled_dataset(root, key, title, data, comp_kw=None):
 def _create_pandas_dataset(fname, root, key, title, data):
     h5py = _check_h5py()
     rootpath = "/".join([root, key])
-    data.to_hdf(fname, key=rootpath)
-    with h5py.File(fname, mode="a") as fid:
-        fid[rootpath].attrs["TITLE"] = "pd_dataframe"
+    if isinstance(fname, h5py.File):
+        # pandas requires full control over the HDF5 file.
+        # The HDF5 file is closed and re-opened.
+        file_name = fname.filename
+        fname.close()
+        # handover control to pandas to write dataset
+        data.to_hdf(file_name, key=rootpath)
+        # Re-open HDF5 file - requires access to internal variable _id
+        fname._id = h5py.File(name=file_name, mode="a")._id
```

Comment:
Any way to do this without messing with private attributes? I worry it will break without warning

Maybe we need to `fname = h5py.File(...)` and `return fname` later. It would be more work to allow `fname` to be modified like this but could avoid the private attribute problem

Reply:
If we replace the `fname` variable then this does not apply for external functionality and the tests break - I tested this in https://github.com/h5io/h5io/pull/146/commits/659c7ead10c36ad8dc7a756d8f2f013792919bf4 . With the extended tests I am also convinced that we are able to identify changes in the underlying libraries early on.